pkg/image/importer/importer.go

```diff
@@ -78,17 +82,28 @@ func (i *ImageStreamImporter) contextImageCache(ctx gocontext.Context) map[manif
 	return cache
 }
 
+type imageStreamImporterCache struct {
+	ImageCache     map[manifestKey]*api.Image
+	LayerSizeCache map[string]int64
+}
+
 // Import tries to complete the provided isi object with images loaded from remote registries.
 func (i *ImageStreamImporter) Import(ctx gocontext.Context, isi *api.ImageStreamImport) error {
-	cache := i.contextImageCache(ctx)
+	if i.digestToLayerSizeCache == nil {
```

Comment:
This should be hidden inside of contextImageCache.  Initialize and return all your caches at once, don't do it at multiple layers.